Core/Foundation/Controller/BaseController.php

```diff
@@ -0,0 +1,246 @@
+     */
+    public function formatResponse($format, Response &$response)
+    {
+        switch ($format) {
+            case 'html':
+                $this->formatHtmlResponse($response);
+                break;
+            case 'json':
+                $this->formatJsonResponse($response);
+                break;
+            case 'xml':
+                throw new DevelopmentErrorException('Not yet supported!');
+            case 'raw':
+                return;
+            case 'none':
+                $this->router->exitNow(); // Break PHP process! Controller action should have already sent its result by itself (file, binary, etc...)
+            default:
+                throw new DevelopmentErrorException('Unknown format.');
+        }
+    }
+
+    /**
+     * This function will format data in an HTML result.
+     *
+     * Most of the time, you should use a template engine to render
+     * the response. The data given by the controller action is available in $response->getContentData(), and once
+     * you rendered the HTML content, you should put it in $response->setContent().
+     *
+     * @param Response $response
+     */
+    abstract protected function formatHtmlResponse(Response &$response);
+
+    /**
+     * This will format data from $response->getContentData() into JSON format.
+     *
+     * @param Response $response
+     */
+    final protected function formatJsonResponse(Response &$response)
+    {
+        $content = $response->getContentData();
+        $response->setContent(json_encode($content, $this->container->make('CoreBusiness:Context')->get('debug') ? JSON_PRETTY_PRINT : 0));
+    }
+
+    /* (non-PHPdoc)
+     * @see \PrestaShop\PrestaShop\Core\Foundation\Controller\ControllerInterface::encapsulateResponse()
+     */
+    public function encapsulateResponse($encapsulation, Response &$response)
+    {
+        switch ($encapsulation) {
+            case 'layout':
+                $this->encapsulateLayout($response);
+                break;
+            case 'nude':
+                $this->encapsulateNudeHtml($response);
+                break;
+            case 'none':
+                return;
+            default:
+                throw new DevelopmentErrorException('Unknown encapsulation.');
+        }
+    }
+
+    /**
+     * This function will encapsulate an HTML content.
+     *
+     * ...into a very smart HTML layout, with the minimum required to be valid HTML document.
+     * If you need more HTML stuff in this mode, override this function in your extended class.
+     *
+     * @param Response $response
+     */
+    protected function encapsulateNudeHtml(Response &$response)
+    {
+        $content = $response->getContent();
+        $content = '<!DOCTYPE html>
+<html lang="en">
+    <head>
+        <meta charset="UTF-8" />
+        <meta name="robots" content="index, follow, all" />
+        <title>Carrément a voir !</title>
+    </head>
+    <body>
+'.$content.'
+    </body>
+</html>';
```

Comment:
No, we can! For performances, I changed this in a heredoc format. A tpl is very heavy for nothing usefull here...